src/dialects/mssql/query-generator.js

```diff
@@ -996,15 +996,38 @@ export class MsSqlQueryGenerator extends AbstractQueryGenerator {
     }
 
     if (options.limit || options.offset) {
-      // TODO: document why this is adding the primary key of the model in ORDER BY
-      //  if options.include is set
+      // TODO: document why this is adding the primary key of the model in ORDER BY if options.include is set
       if (!options.order || options.order.length === 0 || options.include && orders.subQueryOrder.length === 0) {
-        const tablePkFragment = `${this.quoteTable(options.tableAs || model.name)}.${this.quoteIdentifier(model.primaryKeyField)}`;
-        if (!options.order || options.order.length === 0) {
+        let primaryKey = model.primaryKeyField;
+        const tablePkFragment = `${this.quoteTable(options.tableAs || model.name)}.${this.quoteIdentifier(primaryKey)}`;
+        const aliasedAttribute = this._getAliasForFieldFromQueryOptions(primaryKey, options);
+
+        if (aliasedAttribute) {
+          const modelName = this.quoteIdentifier(options.tableAs || model.name);
+          const alias = this._getAliasForField(modelName, aliasedAttribute[1], options);
+
+          primaryKey = new Utils.Col(alias || aliasedAttribute[1]);
```

Comment:
Good question. I'm not even sure if I know what this is doing. I did it because we are doing the same in the abstract query generator 
<img width="713" alt="image" src="https://user-images.githubusercontent.com/79163/185730525-7e54fc20-9546-4fb0-a47d-c6ff31544e52.png">

Reply:
I just pushed an experimental commit to see if this works without the conversion

Reply:
Seems to work just fine without it

Reply:
Interesting, did the queries change that are being executed?

Reply:
Guess I can look up the changes in the queries but just based on the tests there was no apparent change

Reply:
Yeah, this is something I wanted to look into but hadn't had the time yet. I'm just curious if we are missing integration test coverage on this in case of a change in the query

Reply:
Lemme have a look